eng/pipelines/templates/steps/sparse-checkout-for-servicedirectory.yml

```diff
@@ -7,21 +7,25 @@ parameters:
 steps:
   - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
     parameters:
-      ${{ if eq(parameters.ServiceDirectory, 'auto') }}:
-        Paths:
-          - "/*"
-          - ".vscode/"
-      ${{ else }}:
-        Paths:
-          - ${{ if startsWith(parameters.ServiceDirectory, '../') }}:
-            - ${{ replace(parameters.ServiceDirectory, '../', '') }}
-            # this is a workaround for the eng packages that have integration tests that rely on keyvault
-            # todo: transform this to a more generic solution of passed sparse checkout paths
-            - "sdk/security"
-          - ${{ else }}:
-            - "sdk/${{ parameters.ServiceDirectory }}"
-          - ".vscode/"
+      Paths:
+        - "/*"
```

Comment:
If this works I assume we can just remove the sparse-checkout completely and this template.

Reply:
@scbedd said he was finding that just sparse checkout with `/*` was 12s vs 30s on a normal checkout, hence why I've kept it. Regardless I'm going to revert all this anyway in a week or two and go back to the targeted sparse checkout after we move to unified.

Reply:
Ok let's make sure we have an issue tracking that work.

Reply:
https://github.com/Azure/azure-sdk-for-go/issues/24488